### DISCUSSION TITLE
Chips update size and variants

### DISCUSSION THREAD

**Discussion by pihcardozo** [Telefonica/mistica-design#1114] [Proposals]
Seguindo a atualização visual do novo APP da Vivo, vamos precisar do componente Chips com mais variantes:

**- Chips com novo tamanho de 40px (large)**
Vamos precisar criar uma variante do Chips no Figma, no tamanho de 40px de altura.

**- Background transparente. Modo on/off**
Vamos precisar de uma variante com fundo transparente na versão default, isso se aplica para a versão small e large (nova) do Chips.
 
**- Variante com Badge do lado direito**
Vamos precisar de uma variante com Badge na área interna direita do Chips, isso se aplica para a versão small e large (nova) do Chips.

**Comment by yceballost** [comment]
Hola Fabio!

Las discussion están concebidas para ser una zona de debate sobre evoluciones y/o cosas que falten en mística y no una zona de petición sin más, es importante razonar el por qué debería aumentarse el tamaño, para qué casos es necesario el background transparente, etc.

Para el caso del badge, actualmente podrías hacer lo siguiente
https://rebrand.ly/kjuzlk3

**Reply by pihcardozo** [reply]
Hola Ignacio!

En vista de las discusiones y soluciones planteadas en las reuniones, a continuación se presenta una nueva propuesta de actualización del componente Chips.

Teniendo en cuenta el tema de la accesibilidad y al estar muy cerca unos de otros, habrá que aumentar el área táctil de los chips, pasando de 32px a 40px de altura y los márgenes de 8px a 16px y de 12px a 20px. También existe la necesidad de una variante con BG transparente, para la aplicación sobre fondos grises y para mantener la coherencia y actualizar el lenguaje utilizado por otros componentes.

![print](https://user-images.githubusercontent.com/124393542/224791350-a0463e88-161c-4051-93a8-5916801dbb31.png)

**Reply by yceballost** [reply]
Puedes pasar el Figma con ese diseño? me gustaría hacer algunas pruebas en código con las nuevas medidas para saber cómo impactaría en casos actuales

**Reply by pihcardozo** [reply]
https://www.figma.com/file/DrHInkTnSzgACoNX3ULX06/Chips?node-id=0%3A1&t=NABpkFpfs8p9hp4e-1

**Reply by pihcardozo** [reply]
Olá Yayo.
Você poderia criar uma Branch para as especificações do Chips? Já temos o componente pronto para avaliação e para passarmos para a equipe de desenvolvimento.

**Reply by yceballost** [reply]
Hola Fabio! Aún no sabemos si se haremos esta evolución debemos mirarlo con más detalle.

No crees que la versión transparente sobre backgroundAlternative puede dar problemas de accesibilidad? Apenas se puede ver el borde 😥

**Reply by aweell** [reply]
Buenas @pihcardozo,

Cuando deberían usarse las default vs las sin fondo? Es para que las chips funcionen sobre fondos grises (`backgroundAlternative`)? Qué ocurriría en el caso de que estas vayan en un fondo de color o una imagen?

**Reply by pihcardozo** [reply]
Olá @yceballost e @aweell 

A Work usou o chips sobre o background alternative para os níveis 1, selecionei estas telas para demonstração:  

https://www.figma.com/file/DrHInkTnSzgACoNX3ULX06/Chips?node-id=326%3A14029&t=JRHFybsucokfA4OQ-1 

Porém, eles estão aplicados sem o fundo do componente chips, o que o deixa com fundo transparente. Logo ao lado desta demonstração de telas da Work, fiz um exemplo de como ficaria esta variante do chips aplicado sobre diversos fundos.


**Respondendo as perguntas do Alex:**

Cuando deberían usarse las default vs las sin fondo? Es para que las chips funcionen sobre fondos grises (backgroundAlternative)? 
Após fazermos estudos do Chips com fundo transparente, notamos que o melhor a fazer será criar uma variante dele com fundo na cor #f6f6f6, que corresponde a mesma cor do “backgroundAlternative”. Uma sugestão é criar um “scopetokenschips” para que que seja atrelado ao “backgroundAlternative”. Desta forma, teríamos 2 variantes do chips default, uma já existente com o fundo branco e a nova com fundo cinza claro para ser aplicado sobre o “backgroubdAlternative” e não concorrer com os cards que já são brancos. É uma sutileza que impacta bastante o resultado final da interface.


Qué ocurriría en el caso de que estas vayan en un fondo de color o una imagen?"
O Chip Gray pode ser aplicado sobre fundos coloridos ou em imagens, mas neste caso, acreditamos que a versão branca seja a mais "safe" para essa aplicação.

**Reply by aweell** [reply]
Buenas @pihcardozo,

Voy a ir tocando punto a punto:

### Evitar la creación de variantes utilizando contextos

> notamos que o melhor a fazer será criar uma variante dele com fundo na cor #f6f6f6, que corresponde a mesma cor do “backgroundAlternative”

Aquí quizás la solución ideal sea utilizar los contextos

| Property | Context: `default` | Context: `inverse` | Context: `alternative` |
| :------- | :------------------ | :------------------ | :---------------------- |
| `background-color` | `backgroundContainer` | `backgroundContainer` | `backgroundAlternative` |

De esta manera la chip siempre que vaya en un contexto alternative pintará su background del mismo color que el background donde se encuentra.


### Evitar la creación de tokens a no ser que el scope del cambio sea local y no global

> Uma sugestão é criar um “scopetokenschips” para que que seja atrelado ao “backgroundAlternative”. Desta forma, teríamos 2 variantes do chips default, uma já existente com o fundo branco e a nova com fundo cinza claro para ser aplicado sobre o “backgroubdAlternative” e não concorrer com os cards que já são brancos.

La creación de un token sólo debería responder a aquellos cambios que son necesarios entre marcas. Ejemplo: Que la chip se vea blanca en o2 pero azul en telefónica.

Aquí me surge la duda, debería ser un cambio global que las chips en un contexto alternative pinten su background en alternative, o es un cambio que sólo aplicaría a Vivo? 

Desde mi punto de vista en la respuesta a esta pregunta reside la solución a este problema.

### Chips inverse

> O Chip Gray pode ser aplicado sobre fundos coloridos ou em imagens, mas neste caso, acreditamos que a versão branca seja a mais "safe" para essa aplicação.

A día de hoy en la mayoría de los casos donde se esta utilizando imágenes con contenido encima se le está dando el contexto `inverse`, la chips no modificarían su estado por defecto así que entiendo que se pintarían de la manera correcta. 

Aquí quizás enlazo con la issue https://github.com/Telefonica/mistica-design/issues/579, me surge la curiosidad de qué pasaría si tengo un contexto alternative que envuelve a un inverse...

**Reply by yceballost** [reply]
Agree, esta solución sería transversal para todas las marcas usando los contextos. Para todas las marcas aplicaríamos el mismo backgroundAlternative en esta variante. Sería el mejor approach

Me parece bien lo que comenta Cardozo sobre que en aquellos fondos que se salgan de los contextos establecidos se aplicaría el inverse / default y ya.

**Reply by BrunoBiagioniNeto** [reply]
Podemos actualizar la documentación y luego ir con una nueva variante con backgroundAlternative? así todas las marcas tendrían su propia alternativa, ¿sería esa?

**Reply by pihcardozo** [reply]
Olá @yceballost

> Hola Fabio! Aún no sabemos si se haremos esta evolución debemos mirarlo con más detalle.

Podemos seguir com a evolução de 32px para 40px?

**Reply by aweell** [reply]
Buenas @pihcardozo y @BrunoBiagioniNeto,

Se han hecho pruebas de cómo funciona el `height` de las chips a 40px en entornos desktop?

* Debería ser siempre a 40 independientemente del viewportWidth?
* Deberían mantenerse los 32 en desktop y subir 40 solo en mobile?

**Reply by pihcardozo** [reply]
Olá @aweell 

A nossa recomendação é seguirmos somente com a alteração para o chips mobile, sem a necessidade de gastarmos energia neste momento com a evolução para o chips em desktop, já que não é uma dor atual apontada pelos times.

Se no futuro notarmos a necessidade de evolução para desktop, teremos como base pronta a versão mobile.  

Porém, caso queiram manter a mesma lógica de tamanhos iguais que existe hoje, podemos documentar isso.

**Reply by pihcardozo** [reply]
Olá @yceballost 

Conforme discutido na última reunião Vivo-Mística, foi apontada a evolução do chips com Badge interno. Nós já tínhamos discutido isso anteriormente conforme solução dada por você:

> Para el caso del badge, actualmente podrías hacer lo siguiente
https://rebrand.ly/kjuzlk3

A necessidade apontada pela Work é do chips sinalizar alguma ação a ser tomada, como por exemplo: apontar a quantidade de faturas disponíveis para consulta ou pagamento, conforme exemplo:

![Captura de Tela 2023-04-17 às 18 10 51](https://user-images.githubusercontent.com/124393542/232611668-8c6cfde6-6822-4cf3-8d9a-88edd3111c8e.png)

Sendo assim, aguardamos o resultado da analise de impacto global e do Ok de vocês para seguirmos com esta evolução, levando em consideração que esta evolução será uma nova variante do Chips.

**Comment by aweell** [comment]
I created the issue #1160 to cover this discussion
